tb_plugin/torch_tb_profiler/profiler/run_generator.py

```diff
@@ -306,48 +307,73 @@ def __init__(self, all_profile_data):
     def generate_run_profile(self):
         profile_run = DistributedRunProfile()
         profile_run.views.append(consts.DISTRIBUTED_VIEW)
+        profile_run.gpu_info = self._generate_gpu_info()
         profile_run.steps_to_overlap = self._generate_overlap_graph()
         profile_run.steps_to_wait = self._generate_wait_graph()
         profile_run.comm_ops = self._generate_ops_table()
         return profile_run
 
+    def _generate_gpu_info(self):
+        result = OrderedDict()
+        result["metadata"] = {"title": "Device Info"}
+        result["data"] = dict()
+        result["data"]["Process id 123"] = dict()
+        result["data"]["Process id 456"] = dict()
+        result["data"]["Process id 123"]["GPU0"] = {"name": "Tesla V100", "Global Memory": "34G", "Compute Compability": "7.0"}
+        result["data"]["Process id 123"]["GPU1"] = {"name": "Tesla V100", "Global Memory": "34G", "Compute Compability": "7.0"}
+        result["data"]["Process id 456"]["GPU2"] = {"name": "Tesla V100", "Global Memory": "34G", "Compute Compability": "7.0"}
+        result["data"]["Process id 456"]["GPU3"] = {"name": "Tesla V100", "Global Memory": "34G", "Compute Compability": "7.0"}
+        return result
+
     def _generate_overlap_graph(self):
         result = dict()
         result["metadata"] = {"title": "Computaion/Communication Overview", "legends": ["Computation", "Overlapping", "Communication", "Other"], "units": "us"}
         steps_to_overlap = OrderedDict()
+        steps_to_overlap['all'] = OrderedDict()
         for worker,data in self.all_profile_data.items():
-            for i in range(len(data.steps_names)):
+            steps_to_overlap['all'][worker] = [0, 0, 0, 0]
+            step_number = len(data.steps_names)
+            for i in range(step_number):
```

Comment:
use enumerate to get both i and step_name?

Reply:
done